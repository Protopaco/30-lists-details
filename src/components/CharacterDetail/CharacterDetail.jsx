import React, { useState, useEffect } from 'react';
import { BiArrowBack } from "react-icons/bi";
import { fetchCharacter } from '../../services/fetch.js'
import { useParams, Link } from 'react-router-dom'

const CharacterDetail = () => {
    const [chosenCharacter, setChosenCharacter] = useState()
    const [loading, setLoading] = useState(true);
    const { id } = useParams();

    useEffect(async () => {
        try {
            const fetchedCharacter = await fetchCharacter(id)
            setChosenCharacter(fetchedCharacter)
            setLoading(false)
        } catch {
            err => console.log(err)
        }
    }, [])

    return (
        <div>
            {!loading ?
                <div>
                    <Link to='/'>
                        <BiArrowBack />
                    </Link>

                    <div data-testid='characterDetail'>
                        <img src={chosenCharacter.photoUrl} alt={`image of ${chosenCharacter.name}`} />
                        <p>{`Name: ${chosenCharacter.name}`}</p>
                        <p>{`Affiliation: ${chosenCharacter.affiliation}`}</p>
                        <p>ALLIES:</p>
                        <ul>
                            {chosenCharacter.allies.map((ally, index) => {
                                return <p key={index}>{ally}</p>
                            })}
                        </ul>
                        <p>ENEMIES:</p>
                        <ul>
                            {chosenCharacter.enemies.map((enemy, index) => {
                                return <p key={index}>{enemy}</p>
                            })}
                        </ul>
                    </div>
                </div>
                :
                <p>LOADING!</p>
            }
        </div>
    )
}

export default CharacterDetail;