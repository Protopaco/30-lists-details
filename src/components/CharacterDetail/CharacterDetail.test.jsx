// import React from 'react';
// import { act, render, screen, waitFor } from '@testing-library/react';
// import CharacterDetail from './CharacterDetail';
import { BrowserRouter as Router } from 'react-router-dom';

import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail.jsx';

describe('CharacterDetail component', () => {
    afterEach(() => cleanup());
    it('renders CharacterDetail', () => {
        const { asFragment } = render(<Router><CharacterDetail
        /></Router>);
        expect(asFragment()).toMatchSnapshot();
    });
});

// describe('CharacterDetail tests', () => {
    // it('displays CharacterPreview', async () => {

    //     render(<Router>
    //         <CharacterDetail path={'/details/5cf5679a915ecad153ab6900'} />
    //     </Router >);


    //     screen.getByText('LOADING!');

    //     let ul = await screen.findByTestId('characterDetail');

    //     return waitFor(() => {
    //         expect(ul).not.toBeEmptyDOMElement();
    //     })
    // })
// })