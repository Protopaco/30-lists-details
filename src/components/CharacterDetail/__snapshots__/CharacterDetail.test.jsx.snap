// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <div>
    <p>
      LOADING!
    </p>
  </div>
</DocumentFragment>
`;
